explorer/templates.go

```diff
@@ -228,6 +231,35 @@ func makeTemplateFuncMap(params *chaincfg.Params) template.FuncMap {
 			}
 			return str + "remaining"
 		},
+
+		"amountAsDecimalPartsTrimmed": func(v int64, numPlaces int64, useCommas bool) []string {
+			amt := strconv.FormatInt(v, 10)
+			if len(amt) <= 8 {
+				dec := strings.TrimRight(amt, "0")
+				trailingZeros := ""
+				leadingZeros := strings.Repeat("0", 8-len(amt))
+				if ((8 - len(amt)) > 2) {
+					leadingZeros = "00"
+				}
+				return []string{"0", leadingZeros + dec, trailingZeros}
+			}
+			integer := amt[:len(amt)-8]
+			if useCommas {
+				integerAsInt64, err := strconv.ParseInt(integer, 10, 64)
+				if err != nil {
+					log.Errorf("amountAsDecimalParts comma formatting failed. Input: %v Error: %v", v, err.Error())
+					integer = "ERROR"
+					dec := "VALUE"
+					zeros := ""
+					return []string{integer, dec, zeros}
+				}
+				integer = humanize.Comma(integerAsInt64)
+			}
+			dec := strings.TrimRight(amt[len(amt)-8:], "0")
+			dec = string(dec[0:numPlaces])
```

Comment:
There are numerous ways this function can panic.  Please look at all the indexing statements.  And write tests for this function in templates_test.go

Reply:
Some other ideas:

```go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func main() {
	f := func(v int64, numPlaces int64) []string {
		intpart := v / 1e8
		decpart := v % 1e8

		left := strconv.FormatInt(intpart, 10)
		rightTail := fmt.Sprintf("%08d", decpart)
		right := strings.TrimRight(rightTail, "0")
		tail := strings.Repeat("0", len(rightTail)-len(right))
		return []string{left, right, tail}
	}

	fmt.Println(f(4321, 8))
	fmt.Println(f(87654321, 8))
	fmt.Println(f(76543210, 8))
	fmt.Println(f(6543210, 8))
	fmt.Println(f(987654321, 8))
	fmt.Println(f(987654320, 8))
	fmt.Println(f(907654321, 8))
	fmt.Println(f(907654320, 8))
	fmt.Println(f(907654320, 8))
	fmt.Println(f(12345907654320, 8))
}
```

Reply:
Know how to test a sub function within and under the makeTemplateFuncMap function? How do I call that from the unit test?

Reply:
You can either write the function in the package's scope and set it in `makeTemplateFuncMap`, or in the test function you can just run `funcMap := makeTemplateFuncMap(params)` with the params from any network defined in dcrd's chaincfg package.

Reply:
Done